src/guacamole-docker.md

```diff
@@ -0,0 +1,707 @@
+  `MYSQL_USER`.
+
+If any required environment variables are omitted, you will receive an error
+message in the logs, and the image will stop. You will then need to recreate
+the container with the proper variables specified.
+
+(guacamole-docker-mysql-optional-vars)=
+
+#### Optional environment variables
+
+Additional optional environment variables may be used to override Guacamole's
+default behavior with respect to concurrent connection use by one or more
+users. Concurrent use of connections and connection groups can be limited to an
+overall maximum and/or a per-user maximum:
+
+
+`MYSQL_ABSOLUTE_MAX_CONNECTIONS`
+: The absolute maximum number of concurrent connections to allow at any time,
+  regardless of the Guacamole connection or user involved. If set to "0", this
+  will be unlimited. Because this limit applies across all Guacamole
+  connections, it cannot be overridden if set.
+
+  *By default, the absolute total number of concurrent connections is unlimited
+  ("0").*
+
+`MYSQL_DEFAULT_MAX_CONNECTIONS`
+: The maximum number of concurrent connections to allow to any one Guacamole
+  connection. If set to "0", this will be unlimited. This can be overridden on
+  a per-connection basis when editing a connection.
+
+  *By default, overall concurrent use of connections is unlimited ("0").*
+
+`MYSQL_DEFAULT_MAX_GROUP_CONNECTIONS`
+: The maximum number of concurrent connections to allow to any one Guacamole
+  connection group. If set to "0", this will be unlimited.  This can be
+  overridden on a per-group basis when editing a connection group.
+
+  *By default, overall concurrent use of connection groups is unlimited ("0").*
+
+`MYSQL_DEFAULT_MAX_CONNECTIONS_PER_USER`
+: The maximum number of concurrent connections to allow a single user to
+  maintain to any one Guacamole connection. If set to "0", this will be
+  unlimited. This can be overridden on a per-connection basis when editing a
+  connection.
+
+  *By default, per-user concurrent use of connections is unlimited ("0").*
+
+`MYSQL_DEFAULT_MAX_GROUP_CONNECTIONS_PER_USER`
+: The maximum number of concurrent connections to allow a single user to
+  maintain to any one Guacamole connection group. If set to "0", this will be
+  unlimited. This can be overridden on a per-group basis when editing a
+  connection group.
+
+  *By default, per-user concurrent use of connection groups is limited to one
+  ("1")*, to prevent a balancing connection group from being completely
+  exhausted by one user alone.
+
+`MYSQL_AUTO_CREATE_ACCOUNTS`
+: Whether or not accounts that do not exist in the MySQL database will be
+  automatically created when successfully authenticated through other modules.
+  If set to "true" accounts will be automatically created. Otherwise, and by
+  default, accounts will not be automatically created and will need to be
+  manually created in order for permissions within the MySQL database extension
+  to be assigned to users authenticated with other modules.
```

Comment:
Unfortunately, I don't believe so. The options seem to be:

* Document each database separately, leveraging tabs to show only the relevant portions of the documentation.
* Document each database simultaneously, listing all applicable properties at once regardless of which database tab is selected.

I think the tabbed approach provides the best user experience and most readable documentation.

It's possible to leverage something like the `include` directive to include excerpts from a file. For example, if one file contains something like:

```
+++ db-username

The username of the user that Guacamole should use to connect to the database.
This is given as "`guacamole_user`" in the examples given in this chapter.

+++ db-password

The password Guacamole should provide when authenticating with the database.
This is given as "`some_password`" in the examples given in this chapter.

+++
```

then those bits can be included selectively with something like:

    ```{include} name-of-the-file.md
    :start-after: +++ db-username
    :end-before: +++
    ```

We could do that for the common properties.

Reply:
Up to you - I'm not _that_ worried about it, as I don't think the contents will change often, just thought it might save some lines of code and keystrokes if the common text between the JDBC Authentication section and the Docker configuration section could pull from a common location.

Reply:
My preference would be to leave this as it is for now. I'm sure there's a way to reduce duplication across databases and between `guacamole.properties` and Docker, but that definitely feels like a large enough rewrite that it should be a separate thing.